packages/truffle-decoder/lib/decode/calldata.ts

```diff
@@ -177,7 +177,8 @@ function* decodeCalldataStructByPosition(definition: DecodeUtils.AstDefinition,
     //there also used to be code here to add on the "_ptr" ending when absent, but we
     //presently ignore that ending, so we'll skip that
 
-    let decoded = yield* decodeCalldata(memberDefinition, childPointer, info);
+    let decoded = yield* decodeCalldata(memberDefinition, childPointer, info, startPosition);
```

Comment:
Oh so this argument just got lost somewhere, hence the regression.

Reply:
Right -- I didn't bother including it in the static case (because it's meaningless there), and I accidentally used the static code rather than the dynamic code when I combined them.